src/code/z_map_disp.c

```diff
@@ -1,109 +1,1881 @@
+        MapDataRoom* mapDataRoom = &sMapDisp.mapDataScene->rooms[i2];
+
+        if (mapDataRoom->mapId == 0xFFFF) {
+            continue;
+        }
+        // add item to the table if it is a newish value
+        for (i1 = 0; i1 < ROOM_MAX; i1++) {
+            if (sMapDisp.storeyYList[i1] == FLOOR_MIN_Y) {
+                sMapDisp.storeyYList[i1] = mapDataRoom->floorY;
+                break;
+            } else if (fabsf((f32)sMapDisp.storeyYList[i1] - (f32)mapDataRoom->floorY) < 5.0f) {
+                break;
+            }
+        }
+    }
+
+    // sort the table in ascending order
+    for (i2 = 0; i2 < sSceneNumRooms; i2++) {
+        if (sMapDisp.storeyYList[i2] == FLOOR_MIN_Y) {
+            break;
+        }
+        for (i3 = i2 + 1; i3 < sSceneNumRooms; i3++) {
+            if (sMapDisp.storeyYList[i3] == FLOOR_MIN_Y) {
+                break;
+            }
+            if (sMapDisp.storeyYList[i3] < sMapDisp.storeyYList[i2]) {
+                s16 swap = sMapDisp.storeyYList[i2];
+
+                sMapDisp.storeyYList[i2] = sMapDisp.storeyYList[i3];
+                sMapDisp.storeyYList[i3] = swap;
+            }
+        }
+    }
+
+    for (i2 = 0; i2 < sSceneNumRooms; i2++) {
+        MapDataRoom* mapDataRoom = &sMapDisp.mapDataScene->rooms[i2];
+
+        sMapDisp.roomStoreyList[i2] = -1;
+
+        for (storey = 0; storey < sSceneNumRooms; storey++) {
+            if (sMapDisp.storeyYList[storey] != FLOOR_MIN_Y) {
+                if (fabsf((f32)sMapDisp.storeyYList[storey] - (f32)mapDataRoom->floorY) < 5.0f) {
+                    sMapDisp.roomStoreyList[i2] = storey;
+                    break;
+                }
+            }
+        }
+    }
+    sMapDisp.numStoreys = 0;
+    for (i2 = 0; i2 < sSceneNumRooms; i2++) {
+        if (sMapDisp.storeyYList[i2] != FLOOR_MIN_Y) {
+            sMapDisp.numStoreys++;
+        }
+    }
+    sMapDisp.bottomStorey = 0;
+    for (i2 = 0; i2 < ARRAY_COUNT(sCustomBottomStorey); i2++) {
+        if (play->sceneId == sCustomBottomStorey[i2].sceneId) {
+            sMapDisp.bottomStorey = sCustomBottomStorey[i2].bottomStorey;
+        }
+    }
+}
+
+/**
+ * Unused result
+ * @returns the y position to place the boss room skull icon on the pause map.
+ * The result position is the inverse of what it should be, stacking lower rooms above higher ones
+ */
+s32 MapDisp_GetBossIconY(void) {
+    s32 dungeonMapFloorIconPosY[5] = { 67, 81, 95, 109, 123 };
+
+    if ((sMapDisp.mapDataScene == NULL) || (sMapDisp.bossRoomStorey < 0) || (sMapDisp.bossRoomStorey >= 5) ||
+        (sSceneNumRooms == 0)) {
+        return 123;
+    }
+    return dungeonMapFloorIconPosY[sMapDisp.bossRoomStorey];
+}
+
+s16 MapDisp_GetBossRoomStorey(void) {
+    return sMapDisp.bossRoomStorey;
+}
+
+// TransitionActor params test
+s32 MapDisp_IsBossDoor(s32 params) {
+    if (ENDOOR_PARAMS_GET_TYPE((u16)params) == ENDOOR_TYPE_5) {
+        return true;
+    }
+    return false;
+}
```

Comment:
This is a weird situation. To me it seems clear the check tests whether the transition actor is the boss door for a dungeon, but there's no check for the transition actor being DoorShutter specifically, so any transition actor not EnHoll will pass the check. DoorShutter, EnDoor, DoorSpiral can all potentially pass this check; BgOpenShutter neither uses params nor appears in a dungeon. Additionally, EnHoll, DoorShutter, EnDoor all have the same "type" params getter; DoorSpiral doesn't, but has `DOORSPIRAL_GET_TYPE` and `DOORSPIRAL_GET_DIRECTION` packed within the same 3 bits

Reply:
Fixed this to use DoorShutter instead, though I'm curious whether use a bug label or not, and where to put it. Technically no bug ever occurs because with the way the other transition actors appear in dungeons, `MapDisp_IsBossDoor` only reports true on DoorShutter instances, but obviously this would not be true if people try to create new transition types.